tests/components/notify/test_init.py

```diff
@@ -1,28 +1,241 @@
 """The tests for notify services that change targets."""
 
 import asyncio
+import copy
 from pathlib import Path
-from unittest.mock import Mock, patch
+from typing import Any
+from unittest.mock import MagicMock, Mock, patch
 
 import pytest
 import yaml
 
 from homeassistant import config as hass_config
 from homeassistant.components import notify
-from homeassistant.const import SERVICE_RELOAD, Platform
-from homeassistant.core import HomeAssistant
+from homeassistant.components.notify import (
+    DOMAIN,
+    SERVICE_SEND_MESSAGE,
+    NotifyEntity,
+    NotifyEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    SERVICE_RELOAD,
+    STATE_UNAVAILABLE,
+    STATE_UNKNOWN,
+    Platform,
+)
+from homeassistant.core import HomeAssistant, State
 from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
 from homeassistant.helpers.reload import async_setup_reload_service
 from homeassistant.setup import async_setup_component
 
-from tests.common import MockPlatform, async_get_persistent_notifications, mock_platform
+from tests.common import (
+    MockConfigEntry,
+    MockEntity,
+    MockModule,
+    MockPlatform,
+    async_get_persistent_notifications,
+    mock_integration,
+    mock_platform,
+    mock_restore_cache,
+)
+
+TEST_KWARGS = {"message": "Test message"}
+
+
+class MockNotifyEntity(MockEntity, NotifyEntity):
+    """Mock Email notitier entity to use in tests."""
+
+    send_message_mock_calls = MagicMock()
+
+    def send_message(
+        self,
+        message: str | None = None,
+        title: str | None = None,
+        recipients: list[str] | None = None,
+    ) -> None:
+        """Send a notification message."""
+        self.send_message_mock_calls(
+            message=message, title=title, recipients=recipients
+        )
+
+
+async def test_send_message_service(
+    hass: HomeAssistant, config_flow_fixture: None
+) -> None:
+    """Test mode validation for fan, swing and preset."""
+
+    entity = MockNotifyEntity(name="test", entity_id="notify.test")
+
+    async def async_setup_entry_init(
+        hass: HomeAssistant, config_entry: ConfigEntry
+    ) -> bool:
+        """Set up test config entry."""
+        await hass.config_entries.async_forward_entry_setups(config_entry, [DOMAIN])
+        return True
+
+    async def async_setup_entry_notify_platform(
+        hass: HomeAssistant,
+        config_entry: ConfigEntry,
+        async_add_entities: AddEntitiesCallback,
+    ) -> None:
+        """Set up test notify platform via config entry."""
+        async_add_entities([entity])
+
+    mock_integration(
+        hass,
+        MockModule(
+            "test",
+            async_setup_entry=async_setup_entry_init,
+        ),
+        built_in=False,
+    )
+    mock_platform(
```

Comment:
Let's try to use `setup_test_component_platform`.

Reply:
platform set up via `async_setup_platform` is not supported. The legacy `notify` service is set via `get_service`  or `async_get_service` or platform setup. The notify entity can only be set up via an entry setup.

Or is it that we need to add an `async_setup_platform` method as well.

Reply:
`setup_test_component_platform` works with config entry too.

https://github.com/home-assistant/core/blob/469b01bd64eae4f3ed0d642aaa8064fe771b9625/tests/common.py#L1698-L1735